chromium_src/chrome/browser/process_singleton_posix.cc

```diff
@@ -1032,12 +1049,13 @@ bool ProcessSingleton::Create() {
     NOTREACHED() << "listen failed: " << base::safe_strerror(errno);
 
   DCHECK(BrowserThread::IsMessageLoopValid(BrowserThread::IO));
-  BrowserThread::PostTask(
-      BrowserThread::IO,
-      FROM_HERE,
-      base::Bind(&ProcessSingleton::LinuxWatcher::StartListening,
-                 watcher_,
-                 sock));
+  sock_ = sock;
+
+  if (atom::Browser::Get()->is_ready()) {
```

Comment:
You can use `BrowserThread::IsMessageLoopValid(BrowserThread::IO)` to avoiding involving `atom::Browser` here.

I'm not sure why `DCHEK` does not work in this file, but you should remove the previous call since it will abort after we fixed the `DCHECK`.